docs/proposals/cwag_ha/cwag_ha_transport_failover.md

```diff
@@ -0,0 +1,104 @@
+# CWAG HA Transport Failover - Design Proposal
+
+*Status: In-Review*\
+*Author: @mpgermano*\
+*Reviewers: @cwf-team*\
+*Last Updated: 07/24*
+
+# Overview
+
+In H1 of 2020, Magma implemented CWAG high availability through an active/standby design. 
+The initial design relied on the external WLC/AP to properly route RADIUS and GRE traffic 
+to the IP of the active gateway. This document outlines a proposal to handle transport 
+failover as a part of the CWAG HA failover mechanism.
+
+## Transport Failover
+
+The HA design currently relies on the WLC/AP to support redundant GRE tunnels and RADIUS servers. 
+It works in the following way:
+
+1. When the active becomes unhealthy, an ICMP drop rule is added to eth1 of the failed active. 
+   The RADIUS server is also stopped for this gateway.
+2. The WLC detects that the primary transport (both GRE and RADIUS) is down.
+3. After a configurable number of seconds, the WLC will begin using the secondary server 
+   and tunnel configured. 
+
+This design is problematic for a couple of reasons:
+
+* Not all WLC/APs support transport redundancy (e.g. Cambium)
+* The failover determination is separate for the RADIUS and GRE transports. This means that
+  control-plane traffic could be sent to the standby while data-plane traffic is routed to 
+  the active (or vice versa).
+
+To fix this, we want to have a single IP that can properly route traffic to whichever gateway
+is currently designated as active. 
+
+### Using a Virtual IP
+
+We can configure a virtual IP address in the NMS as a network level config that will be 
```

Comment:
This actually should be a config that's specified per active-standby CWAG pair right? AFAIK we don't have any API validation that there can only be 1 CWAG per network.

Myabe we should create a new "HA-pair" network entity in configurator and attach the VIP to that entity's config via the API. Thoughts?

Reply:
Yeah that makes sense. HA-pairs seem to be a common enough pattern across gateway components to justify this.